src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,207 @@
+package duke.storage;
+
+import duke.task.*;
+
+import java.io.*;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+
+
+/**
+ * This class handles the storage and retrieval of tasks to and from a file.
+ */
+public class Storage {
+
+    private String filePath;
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param filePath The path of the file to be used for storage.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the file and returns a TaskArray containing the parsed tasks.
+     *
+     * @return A TaskArray containing the loaded tasks.
+     */
+    public TaskArray load() {
+        // Create the folder/file if it doesn't exist
+        boolean createdFolder = createFolder(this.filePath);
+        boolean createdFile =createFile(this.filePath);
+
+        ArrayList<String> result = new ArrayList();
+        ArrayList<String> tobeProcessedArray = scanFile(this.filePath);
```

Comment:
I understand the naming convention here, but I think maybe explicitly specifying the use of an ArrayList may be clearer? Like use `toBeProcessedArrayList` instead.